### PR TITLE
fix(IV Close Button): Added configuration methods to control the display of the close button

### PR DESCRIPTION
**Pull request by thomas-james-1986**
## JIRA
[AAG-2860]

## DESCRIPTION

This adds the ability to disable the close button, show the close button after a delay or show the close button immediately for IV ads on iOS. 

Note: 

In the case of configuring the close button to show after a delay, if an `adShown` event isn't received via the JS bridge the close button will show after a default 12 second delay

## SCREENSHOTS
n/a

[AAG-2860]: https://superawesomeltd.atlassian.net/browse/AAG-2860?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Pod/Classes/UI/Managed/SAManagedAdView.swift [hunk 1]
```diff
@@ -198,6 +198,22 @@ extension SAManagedAdView: WKNavigationDelegate, WKUIDelegate {
                         decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
         decisionHandler(.allow)
     }
+
+    // Error handling
+
+    public func webView(_: WKWebView, didFail: WKNavigation!, withError: Error) {
```
Comment (thomas-james-1986) [inline Pod/Classes/UI/Managed/SAManagedAdView.swift:204]: @gunhansancar I've added some initial error handling and thought maybe we should discuss more between us. As per your suggestion, we could match the Android error handling like this if we want to restrict the errors we handle:

```
        let error = withError as NSError
        if error.code == NSURLErrorCannotFindHost {
            onEvent(event: .adFailedToLoad)
        }
```

File: Pod/Classes/UI/Managed/SAManagedAdViewController.swift [hunk 2]
```diff
@@ -86,6 +88,44 @@ import WebKit
...
+            closeButtonFallbackTimer = Timer.scheduledTimer(timeInterval: 1,
+                                                            target: self,
+                                                            selector: #selector(closeButtonVisibilityFallback),
+                                                            userInfo: nil,
+                                                            repeats: true)
+        case .visibleImmediately:
+            button.isHidden = false
+        case .hidden:
+            button.isHidden = true
+        }
```
Comment (me2k) [inline Pod/Classes/UI/Managed/SAManagedAdViewController.swift:104]: Nice and neat 😁

File: Pod/Classes/UI/Managed/SAManagedAdViewController.swift [hunk 3]
```diff
@@ -109,12 +149,23 @@ import WebKit
...
+        viewableDetector?.cancel()
+        viewableDetector = nil
+    }
 }
 
 extension SAManagedAdViewController: AdViewJavaScriptBridge {
     func onEvent(event: AdEvent) {
         callback?(placementId, event)
 
+        if (event == .adShown && config.closeButtonState == .visibleWithDelay) {
```
Comment (me2k) [inline Pod/Classes/UI/Managed/SAManagedAdViewController.swift:165]: Nice!

**Review by me2k — APPROVED** [review]
Great job Tom!

**Pull request merged**